### PR TITLE
feat(datagrid): Design-system & start typescript migration

### PR DESCRIPTION
**Pull request by gjou-tlnd**
**What is the problem this PR is trying to solve?**

- Use design-tokens & design-system components when possible
- Boot typescript config

Quality bar & header renderer will be updated in another PR

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/datagrid/src/components/DataGrid/DataGrid.scss [hunk 1]
```diff
@@ -80,17 +87,17 @@ $td-datagrid-skeleton-height: 9.6rem !default;
...
-		font-family: $td-body-font-family;
+		background-color: tokens.$coral-color-neutral-background-medium;
+		font: tokens.$coral-paragraph-m;
 	}
 
 	:global(.ag-cell) {
-		border-right: 0.1rem solid $alto;
-		border-bottom: 0.1rem solid $alto;
+		border-right: tokens.$coral-color-neutral-border-weak tokens.$coral-border-s-solid;
+		border-bottom: tokens.$coral-color-neutral-border-weak tokens.$coral-border-s-solid;
```
Comment (frassinier) [inline packages/datagrid/src/components/DataGrid/DataGrid.scss:96]:
```suggestion
		border-right: tokens.$coral-border-s-solid tokens.$coral-color-neutral-border-weak;
		border-bottom: tokens.$coral-border-s-solid tokens.$coral-color-neutral-border-weak;
```

File: packages/datagrid/src/components/DataGrid/DataGrid.scss [hunk 2]
```diff
@@ -1,35 +1,42 @@
+@use '~@talend/design-tokens/lib/_tokens.scss';
```
Comment (frassinier) [inline packages/datagrid/src/components/DataGrid/DataGrid.scss:1]:
```suggestion
@use '~@talend/design-tokens/lib/tokens';
```

File: packages/datagrid/src/components/DataGrid/DataGrid.component.js [hunk 3]
```diff
@@ -330,7 +330,11 @@ export default class DataGrid extends React.Component {
 	render() {
 		let content;
 		if (this.props.loading) {
-			content = <Skeleton name="talend-table" type={Skeleton.TYPES.icon} />;
+			content = (
+				<div className={theme['td-grid-loader']}>
+					<Icon name="talend-table" />
+				</div>
```
Comment (sgangnant) [inline packages/datagrid/src/components/DataGrid/DataGrid.component.js:336]: Do we no longer have a shorter-hand version of this with the design system ?

Comment (Benrajalu) [reply packages/datagrid/src/components/DataGrid/DataGrid.component.js:336]: Nope. And the notion of a skeleton icon is _weird_. In the future we'll hopefully provide loaders.

**Review by romainseb — APPROVED** [review]

**Review by sgangnant — APPROVED** [review]

**Pull request merged**